packages/block-editor/src/components/link-control/test/index.js

```diff
@@ -2056,6 +2056,25 @@ describe( 'Addition Settings UI', () => {
 			} )
 		);
 	} );
+
+	it( 'should show tooltip with full URL alongside filtered display', async () => {
+		const user = userEvent.setup();
+		const url =
+			'http://www.wordpress.org/wp-content/uploads/a-document.pdf';
+		render( <LinkControl value={ { url } } /> );
+
+		const link = screen.getByRole( 'link' );
+
+		expect( link ).toHaveTextContent( /a-document.pdf/i );
```

Comment:
We can probably simplify by going with a string instead of a regular expression here.

```suggestion
		expect( link ).toHaveTextContent( 'a-document.pdf' );
```